pulp_container/app/registry_api.py

```diff
@@ -620,35 +622,54 @@ def partial_update(self, request, path, pk=None):
         Process a chunk that will be appended to an existing upload.
         """
         _, repository = self.get_dr_push(request, path)
+        upload = get_object_or_404(models.Upload, repository=repository, pk=pk)
         chunk = request.META["wsgi.input"]
-        if "Content-Range" in request.headers or "digest" not in request.query_params:
-            whole = False
-        else:
-            whole = True
+        if range_header := request.headers.get("Content-Range"):
+            found = self.content_range_pattern.match(range_header)
+            start = int(found.group(1))
+            end = int(found.group(2))
+            length = end - start + 1
 
-        if whole:
-            start = 0
         else:
-            content_range = request.META.get("HTTP_CONTENT_RANGE", "")
-            match = self.content_range_pattern.match(content_range)
-            start = 0 if not match else int(match.group("start"))
-
-        upload = get_object_or_404(models.Upload, repository=repository, pk=pk)
+            # is this header always present?
+            length = int(request.headers["Content-Length"])
+            start = 0
 
-        chunk = ContentFile(chunk.read())
         with transaction.atomic():
             if upload.size != start:
                 raise Exception
 
-            upload.append(chunk, upload.size)
-            upload.size += chunk.size
+            # if more chunks
+            if range_header:
+                upload.append(chunk, upload.size)
+            else:
+                # 1 chunk
+                # do not add to the upload, create artifact right away
+                with NamedTemporaryFile("ab") as temp_file:
+                    temp_file.write(chunk.read())
+                    temp_file.flush()
+
+                    uploaded_file = PulpTemporaryUploadedFile.from_file(
+                        File(open(temp_file.name, "rb"))
+                    )
+                try:
+                    artifact = Artifact.init_and_validate(uploaded_file)
+                    artifact.save()
```

Comment:
This artifact we created here does not seem to be related to the upload we patched. So it is created as an orphan right? Don't we need to associate it with an new repo-version right away?